R/loadPEPdb.R

```diff
@@ -0,0 +1,64 @@
+############################################################################################## 
+# PEP-loading function 
+
+##############################################################################################
+
+
+#' Helper function that allows LOLA to read PEP-formatted databases 
+#' 
+#' @param configFolder     folder where the config.yaml file is located
+#' @param configName       Name of the config.yaml file 
+#' @param useCache            uses simpleCache to cache and load the results    
+#' @return PEPdB contains database location, samples annotation sheet, .yaml config file 
+#' and the regions Granges list 
+#' @export
+#' @examples 
+#' ConfigPath = system.file("extdata", "hg19", "ucsc_examplePEP", "project_config.yaml", package="LOLA")
+#' PEPdb = loadPEPdb_CACHE(configLocation=ConfigPath)
+#' 
+
+
+loadPEPdb = function(configFolder, configName, useCache=TRUE){
+	configLoc = file.path(path=paste0(configFolder, configName))
+	if (file.exists(configLoc)) {
+		# Use pepr to read in the PEP metadata
+		pepObject = pepr::Project(file = configLoc)
+		configFile = pepr::config(pepObject)
+		samplesAnnotation = pepr::samples(pepObject)
+		# Rename columns to make annotation LOLA compatible 
+		colnames(samplesAnnotation)[colnames(samplesAnnotation)=="file_name"] = "filename"
+		colnames(samplesAnnotation)[colnames(samplesAnnotation)=="cell_type"] = "cellType"
+		colnames(samplesAnnotation)[colnames(samplesAnnotation)=="exp_protocol"] = "collection"
+		colnames(samplesAnnotation)[colnames(samplesAnnotation)=="data_source"] = "dataSource"
+		# Output the samples annotation and make it a dataframe so that we can lapply through each file path
```

Comment:
No lines longer than 80 characters.

https://www.bioconductor.org/developers/how-to/coding-style/